### ISSUE TITLE
Question about waiting for 127.0.0.1:8080 to be ready

### ISSUE THREAD

**Issue by Nepoxx** [hermanjunge/k8s-do-tf#27]
My setup always gets stuck on the following line for a few minutes (>= 8):

`"until $(curl --output /dev/null --silent --head --fail http://127.0.0.1:8080); do printf '.'; sleep 5; done",`

I'm not totally familiar with Kubernetes (I've used it a lot on GCE but never my own), could someone explain why we keep curling this and what it represents? Why does it take so long?

**Comment by 5tefan** [comment]
I'm experiencing the long wait as well. I'm experiencing up to 20h elapsed which is clearly unacceptable. 

I suspect you're probably using a new version of hyperkube, perhaps hypyercube_version == v1.5.4_coreos.0. You'll notice the default in deploy.tf is 1.3.6, and from the coreos docs for that version[1]:

> ## Create Namespaces
>The Kubernetes Pods that make up the control plane will exist in their own namespace. We need to create this namespace so these components are discoverable by other hosts in the cluster.
>
>Note: You will only need to do this once per-cluster. If deploying multiple master nodes, this step needs to happen only once.
>
>First, we need to make sure the Kubernetes API is available (this could take a few minutes after starting the kubelet.service)
>
>$ curl http://127.0.0.1:8080/version
>A successful response should look something like:
>
>{
>  "major": "1",
>  "minor": "1",
>  "gitVersion": "v1.1.7_coreos.2",
>  "gitCommit": "388061f00f0d9e4d641f9ed4971c775e1654579d",
>  "gitTreeState": "clean"
>}
>Now we can create the kube-system namespace:

>$ curl -H "Content-Type: application/json" -XPOST -d'{"apiVersion":"v1","kind":"Namespace","metadata":{"name":"kube-system"}}' "http://127.0.0.1:8080/api/v1/namespaces"

For more recent version of hyperkube, the coreos docs[2] no longer mention creating the kube-system namespace. This project needs to be updated since setup for hyperkube has changed in newer versions. 

TL;DR: Immediate fix: use a lower version of hyperkube (closer to this project's default 1.3.6, which should work). Proper fix: we need a PR updating deployment process (documented at [2]) for recent versions of hyperkube.* 

[1] https://coreos.com/kubernetes/docs/1.3.6/deploy-master.html
[2] https://coreos.com/kubernetes/docs/1.5.4/deploy-master.html
*I'll work on this today
